pkg/cmd/autocomplete/autocomplete.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package autocomplete
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var supportedShells = []string{"bash", "zsh", "fish", "powershell"}
+
+var AutoCompleteCmd = &cobra.Command{
+	Use:   fmt.Sprintf("autocomplete [%s]", strings.Join(supportedShells, "|")),
+	Short: "Adds a completion script for your shell environment",
+	Args:  cobra.ExactArgs(1),
+}
+
+func init() {
+	AutoCompleteCmd.RunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
Why is this in `init()` and not directly in `var AutocompleteCmd`?

Reply:
because `AutoCompleteCmd.RunE` references `SetupAutocompletionForShell` which again references `AutoCompleteCmd` causing issues

Reply:
The cycle is still a bit problematic.

I suggest that `SetupAutocompletionForShell` takes `rootCmd *cobra.Cobra` as an argument which can be passed as `cmd.Root()` from `RunE` and as `autocomplete.AutocompleteCmd.Root()` from the config package.

(`DetectShellAndSetupAutocompletion` would also have the same arg)

Reply:
okay this is better